src/test/java/htsjdk/samtools/CRAMComplianceTest.java

```diff
@@ -265,6 +295,80 @@ public void testBAMThroughCRAMRoundTrip() throws IOException, NoSuchAlgorithmExc
         }
     }
 
+//    @Test
+//    public void testCompressionWithAllAmbiguityCodes() {
+//        createAmbiguityCodeTestSAM();
+//    }
+
+    @Test
+    public void testRoundTripSAMWithAllAmbiguityCodesThroughCRAM() throws IOException {
+        final File inputSamFile = new File("src/test/resources/htsjdk/samtools/cram/", "allReferenceBases.sam");
+        final File inputReference = new File("src/test/resources/htsjdk/samtools/cram/", "allReferenceBases.fasta");
+        final SAMFileHeader samHeader = getFileHeader(inputSamFile, inputReference);
+
+        List<SAMRecord> originalSAMRecords = getSAMRecordsFromFile(inputSamFile, inputReference);
+        final File tempCRAMFile = File.createTempFile("roundTripAmbiguityCodes", CramIO.CRAM_FILE_EXTENSION);
+        tempCRAMFile.deleteOnExit();
+
+        writeRecordsToFile(
+                originalSAMRecords,
+                tempCRAMFile,
+                inputReference,
+                samHeader
+        );
+
+        List<SAMRecord> roundTripRecords = getSAMRecordsFromFile(tempCRAMFile, inputReference);
+        for (int i = 0; i < originalSAMRecords.size(); i++) {
+            final SAMRecord originalSAMRec = originalSAMRecords.get(i);
+            final SAMRecord roundTripCRAMRec = roundTripRecords.get(i);
+
+            //if (!originalSAMRec.equals(roundTripRecords.get(i))) {
+            if (!Arrays.equals(originalSAMRec.getReadBases(), roundTripCRAMRec.getReadBases())) {
+                System.out.println(originalSAMRec.getSAMString() + roundTripCRAMRec.getSAMString());
+                originalSAMRec.equals(roundTripCRAMRec);
+            }
+            //Assert.assertEquals(originalSAMRecords.get(i), roundTripRecords.get(i));
+        }
+    }
+
+    private void createAmbiguityCodeTestSAM() {
```

Comment:
Seems like this needs a comment explaining what that it was used to generate some of the test files.  Can you not just call this method at test time instead of checking in the test files? Or did you need to run samtools to generate the cram version of it?